drivers/include/cst816s.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_cst816s Cst816S touch screen driver
+ *
+ * @ingroup     drivers_sensors
+ * @brief       Device driver interface for the Hynitron CST816S touch screen
+ *
+ * The CST816S is a touch sensor from Hynitron with integrated gesture
+ * detection. It is able to measure both the position of a single finger and a
+ * number of basic gestures. The PineTime board has one of these for the touch
+ * screen.
+ *
+ * Documentation about the specifics is very limited and most of this driver is
+ * based on experimenting with the chip and from community effort on the
+ * PineTime.
+ *
+ * Two things about the driver are noteworthy:
+ *  1. It only responds to I2C commands after an event, such as a touch
+ *     detection. Do not expect it to respond on init. Instead after a touch
+ *     event, it will assert the IRQ and respond to I2C reads for a short time.
+ *  2. While it should be able to detect multiple finger events, this version of
+ *     the chip always returns only a single finger event and a gesture.
+ *
+ * Reading the display data multiple times during a single event will return the
+ * last sampled finger position.
+ *
+ * @{
+ * @file
+ * @brief       Device driver interface for the CST816S touch screen
+ *
+ * @author      koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef CST816S_H
+#define CST816S_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief cst816s_t forward declaration
+ */
+typedef struct _cst816s cst816s_t;
+
+/**
+ * @brief callback definition
+ */
+typedef void (*cst816s_irq_cb_t)(cst816s_t *dev, void *arg);
```

Comment:
Why do you need `dev` as first argument ? It doesn't seem to be useful. The gpio irq callback could just call `dev->cb(arg)` the same.

Reply:
Sure, seems to work good enough for the other touch screen driver :)